torchtune/models/flamingo/_encoder.py

```diff
@@ -19,22 +18,20 @@ class FlamingoProjectionHead(nn.Module):
     For example, nn.Sequential(CLIP(), FlamingoProjectionHead()).
 
     Args:
-        layer (nn.Module): Transformer Decoder layer
-        num_layers (int): Number of Transformer Decoder layers
+        layers (nn.Module): Transformer Decoder layers
         output (nn.Module): Output linear layer. Input dim is
             (num_hidden + 1) * encoder_dim and output is decoder_dim.
         num_hidden_inputs (int): Number of expected hidden state inputs
     """
 
     def __init__(
         self,
-        layer: nn.Module,
-        num_layers: int,
+        layers: nn.Module,
         output: nn.Module,
         num_hidden_inputs: int = 0,
     ) -> None:
         super().__init__()
-        self.layers = _get_clones(layer, num_layers)
+        self.layers = nn.ModuleList(layers)
```

Comment:
Shouldn't layers be `List[nn.Module]` type then?

Reply:
please look at the transformer, that supports all cases (List[nn.Module], nn.ModuleList, nn.module: https://github.com/pytorch/torchtune/blob/30b8519e86d2094879576e16ab1f8d21a6f6f9f7/torchtune/modules/transformer.py#L353

I am fine if in flamingo it only supports nn.ModuleList though, but i prefer the consistency